influxdb3/point.go

```diff
@@ -248,6 +248,22 @@ func (p *Point) MarshalBinaryWithDefaultTags(precision lineprotocol.Precision, d
 	enc.SetPrecision(precision)
 	enc.StartLine(p.Values.MeasurementName)
 
+	// N.B. Some customers have requested support for newline and tab chars in tag values (EAR 5476)
+	// Though this is outside the lineprotocol specification, it was supported in
+	// previous GO client versions.
+	tagEscapes := map[string]string{
+		"\n": "\\n",
+		"\t": "\\t",
+	}
+
+	replacer := func(reps map[string]string, s string) string {
+		result := s
+		for key, val := range reps {
+			result = strings.ReplaceAll(result, key, val)
+		}
+		return result
+	}
```

Comment:
What do you thinks about using package level `Replacer`?

Something like:

```golang
var replacer = strings.NewReplacer(
	"\n", "\\n",
	"\t", "\\t",
)
```

and then use:

```golang
		if value, ok := p.Values.Tags[tagKey]; ok {
			enc.AddTag(tagKey, replacer.Replace(value))
		} else {
			enc.AddTag(tagKey, replacer.Replace(defaultTags[tagKey]))
		}
```

Reply:
Aha... I was unaware that the `Replacer` type already exists in the `strings` package.  Updated.